src/pybind/mgr/cephadm/module.py

```diff
@@ -2176,6 +2176,12 @@ def _apply_service(self, spec: ServiceSpec) -> bool:
 
         daemon_type = spec.service_type
         service_name = spec.service_name()
+        # if 4 hosts, only deploy 3 mons by default rather than 4
+        if service_name == 'mon' and spec.placement.default:
```

Comment:
What about making this a bit more generic? 

There are multiple cases that need to be taken care of.

* one node -> one mon
* two nodes -> one mon
* three nodes -> three mons
* four nodes -> three mons
* five nodes -> five mons
* more than five -> five mons

The rule is probably `approximate to 3 and 5 but avoid even numbers`

Reply:
1 -> 1, 3 -> 3, 5 -> 5 and more than 5 -> 5 already work as is but I'll add another case for 2 -> 1.